design/noobaa/system-update.md

```diff
@@ -0,0 +1,21 @@
+♜ [Rook NooBaa Design](README.md) /
+# System Update
```

Comment:
This isn't exactly related to this doc, but just a thought to consider... A fundamental behavior of CRDs to understand is that the operator can't tell the difference between creating a new CR and processing the CR when the operator is restarted. Either way, the operator processes a "Create" event for the CR. Only if the operator is running when the CR is updated will the "Update" event be raised and the operator will be able to see what changed from the previous version of the CR. 

Therefore, handling of CR creation or update should always be idempotent. The operator needs to look at the desired state and see what needs to change compared to the actual state of the system. Just don't assume the "Update" method will always be available to tell the differences in what changed.

Reply:
Thanks for clarifying this.
I changed also in the design to combine the sections for Create and Update.
Resolving.